### PR TITLE
Cloudflare rules and queries

### PR DESCRIPTION
**Pull request by vu-socprime**
## Proposed Changes

  - add Cloudflare detection rules
  - add Cloudflare hunting queries

### CODE REVIEW

File: Detections/Cloudflare/CloudflareBadClientIp.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+id: a7ce6135-9d55-4f14-b058-adc2e920a4fa
+name: Cloudflare - Bad client IP
+description: |
+  'Detects requests from IP with bad reputation index.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: CloudflareDataConnector
+    dataTypes:
```
Comment (v-maudan) [inline Detections/Cloudflare/CloudflareBadClientIp.yaml:8]: @vu-socprime , Detection and hunting query operates on a Parser Cloudflare , dataTypes should be the Parser name and not the table name. Please update dataTypes  as Cloudflare in all files.

**Review by v-maudan — CHANGES_REQUESTED** [review]

**Review by shainw — COMMENTED** [review]
Queries are approved, but the dataType value needs to be updated first.

File: Detections/Cloudflare/CloudflareBadClientIp.yaml (hunk 1 above)
Comment (vu-socprime) [reply Detections/Cloudflare/CloudflareBadClientIp.yaml:8]: updated

**Review by v-jayakal — APPROVED** [review]

**Comment by v-maudan** [comment]
@vu-socprime , I have added CustomTable JSON with the Parser name(Cloudflare.json) [here ](https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables)in order to pass KQL validations  , Could you please check once and if any changes required raise another PR for same. Thanks!

**Pull request merged**